File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.cpp
class Solution
{
    public:
        string removeStars(string s)
        {
            int n = s.length();
            string ans = "";
            int count = 0;
            for (int i = n - 1; i >= 0; i--)
            {
                if (s[i] == '*')
                {
                    count++;
                }
                else
                {
                    if (count > 0)
                    {
                        count--;
                    }
                    else
                    {
                        ans += s[i];
                    }
                }
            }
            reverse(ans.begin(), ans.end());
            return ans;
        }
};

File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.java
class Solution {
    public String removeStars(String s) {
        String ans="";
        int n= s.length();
        int count=0;
        for(int i=n-1; i>=0; i--){
            if(s.charAt(i)=='*'){
                count++;
            }
            else{
                if(count==0){
                    ans=  s.charAt(i)+ ans;
                }
                else{
                 count--;
                }
            }
        }
        return ans;
    }
}
